ssh/browser/browser.go

```diff
@@ -0,0 +1,213 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Builds and serves the static site that is generated for the browser/Wasm
+// Tailscale SSH client. Can be run in 3 modes:
+// - dev: builds the site and serves it. JS and CSS changes can be picked up
+//   with a reload.
+// - build: builds the site and writes it to dist/
+// - serve: serves the site from dist/ (embedded in the binary)
+package main
+
+import (
+	"embed"
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strconv"
+
+	esbuild "github.com/evanw/esbuild/pkg/api"
+)
+
+var (
+	dev   = flag.Bool("dev", false, "Run in dev build and serve mode")
+	build = flag.Bool("build", false, "Run in production build mode (generating static assets)")
+	serve = flag.Bool("serve", false, "Run in production serve mode (serving static assets)")
+	addr  = flag.String("addr", ":9090", "address to listen on")
+)
+
+func main() {
+	flag.Parse()
+
```

Comment:
```
if sum(*dev, *build, *serve) != 1 {
    log.Fatalf("exactly one of --dev, --build, --serve expected")
}
```

?

Reply:
Ended up going with a `cmd` arg instead of three mutually exclusive flags, that seemed to more clearly communicate intent.